### ISSUE TITLE
Typescript: createAnimatedComponent() type definition raises type error

### ISSUE THREAD

**Issue by helixbass** [software-mansion/react-native-reanimated#1718]
## Description

When typechecking project with Reanimated v2, the Typescript compiler (`tsc`) flags the type definition for `createAnimatedComponent()` with `Type 'P' does not satisfy the constraint 'object'.`

This makes sense because in this definition there is no constraint on `P` but `AnimateProps` has the constraint `P extends object`:
```
export function createAnimatedComponent<P>(
  component: ComponentClass<P>
): ComponentType<AnimateProps<P>>;
```

So presumably the fix is to declare a `P extends object` type constraint on `P` (this does resolve the type error in my testing):
```
export function createAnimatedComponent<P extends object>(
  component: ComponentClass<P>
): ComponentType<AnimateProps<P>>;
```

## Steps To Reproduce

Run `tsc` against a Typescript project using Reanimated v2 (2.0.0-rc.3) and where `tsconfig.json` is not configured to skip checking `node_modules/` type definition files (eg doesn't include `skipLibCheck: true`)

### Expected behavior

No type error is raise

### Actual behavior

Type error `Type 'P' does not satisfy the constraint 'object'.` is raised

## Package versions

- React: 16.13.1
- React Native: 0.63.3
- React Native Reanimated: 2.0.0-rc.3
- NodeJS: 14.15.3
- Typescript: 3.9.7

**Comment by helixbass** [comment]
I see this would be fixed by #1697

**Comment by helixbass** [comment]
...and I see this is already fixed on `master` via #1698, closing
